nomad/encrypter.go

```diff
@@ -155,12 +155,33 @@ const keyIDHeader = "kid"
 // SignClaims signs the identity claim for the task and returns an
 // encoded JWT with both the claim and its signature
 func (e *Encrypter) SignClaims(claim *structs.IdentityClaims) (string, error) {
-	e.lock.RLock()
-	defer e.lock.RUnlock()
 
-	keyset, err := e.activeKeySetLocked()
+	getActiveKeyset := func() (*keyset, error) {
+		e.lock.RLock()
+		defer e.lock.RUnlock()
+		keyset, err := e.activeKeySetLocked()
+		return keyset, err
+	}
```

Comment:
Sorry for the after-the-fact drive-by review:

Looks like we could replace `activeKeySetLocked` with this func as the only other place the key is read is `Encrypt` and it similarly only needs the lock for the duration of fetching the key.

The fact neither reader of this field need the lock other than for this field signals to me that the lock should be encapsulated in the getter and not the caller's concern.

Not a big deal though.

Reply:
That's a good catch. `activeKeySetLocked` ends up holding onto the lock longer than it needs to as well -- it doesn't need to hold the lock while querying the state store. I'll open another quickie PR for that.

Reply:
Follow-up PR: https://github.com/hashicorp/nomad/pull/15026